app/controllers/allocations_controller.rb

```diff
@@ -46,49 +47,44 @@ def edit
 
     @current_pom = current_pom_for(nomis_offender_id_from_url)
   end
-  # rubocop:enable Metrics/MethodLength
 
   def confirm
     @prisoner = offender(nomis_offender_id_from_url)
     @pom = PrisonOffenderManagerService.get_pom(
       active_caseload,
       nomis_staff_id_from_url
     )
+    @event = :allocate_primary_pom
+    @event_trigger = :user
   end
 
-  # rubocop:disable Metrics/MethodLength
-  # rubocop:disable Metrics/LineLength
-  def create
-    offender = offender(allocation_params[:nomis_offender_id])
-    pom = PrisonOffenderManagerService.get_pom(
+  def confirm_reallocation
+    @prisoner = offender(nomis_offender_id_from_url)
+    @pom = PrisonOffenderManagerService.get_pom(
       active_caseload,
-      allocation_params[:nomis_staff_id]
+      nomis_staff_id_from_url
     )
+    @event = :reallocate_primary_pom
```

Comment:
There may be a better way to do this ...
We need to make sure that we know when we are dealing with an allocation or reallocation. Both require a 'confirm' step so now this has been split into confirm(allocation) and confirm_reallocation. The naming could be changed ...